crates/pathfinder/src/state/merkle_tree.rs

```diff
@@ -51,10 +51,56 @@ use pathfinder_storage::merkle_tree::{
     NodeStorage, PersistedBinaryNode, PersistedEdgeNode, PersistedNode, RcNodeStorage,
 };
 use rusqlite::Transaction;
+use serde::Serialize;
 use stark_hash::StarkHash;
 use std::ops::ControlFlow;
 use std::{cell::RefCell, rc::Rc};
 
+/// Lightweight representation of [BinaryNode]. Only holds left and right hashes.
+#[derive(Debug, PartialEq, Serialize)]
```

Comment:
Is `derive(Serialize)` needed here? If not then I can see two solutions here:
- add the same derive to `EdgeProofNode` and `ProofNode`, and see if serde annotation ( `tag = "type"`) works fine with `ProofNode` and then remove the manual serialization impl in `get_proof.rs`
- or just remove the derive here and leave the impl in `get_proofs.rs`.

Reply:
bbc8a3f51512f0f31b0767febf953e6939652423